### PR TITLE
add grpc-ecosystem/openapiv2 plugin

### LINKED ISSUE #250: Plugin request for Buf Schema Registry - openapiv2
## Mandatory

**Where is the source code for the plugin?**

https://github.com/grpc-ecosystem/grpc-gateway/tree/main/protoc-gen-openapiv2

## Optional

**Does the plugin have a valid semver version?**

Yes, latest version is v2.14.0.

**Do you think this plugin will be compatible with the BSR go module proxy or npm registry?**

Not sure, it only generates a yaml file.

### PR DESCRIPTION
**Pull request by pkwarren**
Add the protoc-gen-openapiv2 plugin from grpc-gateway.

Fixes #250.

### CODE REVIEW

File: tests/testdata/buf.build/grpc-ecosystem/openapiv2/v2.15.0/petapis/plugin.sum [hunk 1]
```diff
@@ -0,0 +1 @@
+h1:kbnot+zVGjxK+SAIaEjtKgvlHNA+4RVmP6bjvSYh9JA=
```
Comment (pkwarren) [inline tests/testdata/buf.build/grpc-ecosystem/openapiv2/v2.15.0/petapis/plugin.sum:1]: Even though we don't have HTTP annotations in the input files, I verified that there is swagger generated for the services and types:
```
$ tree tests/testdata/buf.build/grpc-ecosystem/openapiv2/v2.15.0/*/gen
tests/testdata/buf.build/grpc-ecosystem/openapiv2/v2.15.0/eliza/gen
└── buf
    └── connect
        └── demo
            └── eliza
                └── v1
                    └── eliza.swagger.json
tests/testdata/buf.build/grpc-ecosystem/openapiv2/v2.15.0/petapis/gen
└── pet
    └── v1
        └── pet.swagger.json
```

We may want to add a more comprehensive test that has the http annotations in a follow up (especially for grpc-gateway which I think doesn't emit anything if they aren't defined).

Comment (johanbrandhorst) [reply tests/testdata/buf.build/grpc-ecosystem/openapiv2/v2.15.0/petapis/plugin.sum:1]: The gateway can be coerced to generate mappings even in the absence of annotations: https://github.com/grpc-ecosystem/grpc-gateway/blob/main/protoc-gen-grpc-gateway/main.go#L37

Comment (pkwarren) [reply tests/testdata/buf.build/grpc-ecosystem/openapiv2/v2.15.0/petapis/plugin.sum:1]: Great! I'll look to set that option for the grpc-gateway test execution then.

**Review by mfridman — APPROVED** [review]
lgtm. Great work 🎉

**Pull request merged**